src/entry_point.py

```diff
@@ -136,9 +142,58 @@ def __call__(self, parser, namespace, values, option_string=None):
         "in the environment specified by --prefix",
     )
 
+    subcommands = p.add_subparsers(dest="command")
+    uninstall_subcommand = subcommands.add_parser(
+        "uninstall",
+        description="Uninstalls a conda directory and all environments inside the directory.",
+    )
+    uninstall_subcommand.add_argument(
+        "--prefix",
+        action="store",
+        required=True,
+        help="Path to the conda directory to uninstall.",
+    )
+    uninstall_subcommand.add_argument(
+        "--conda-clean",
+        action="store_true",
+        required=False,
+        help=(
+            "Run conda --clean --all to remove package caches outside the installation directory."
+            " This is only useful when pkgs_dirs is set in a .condarc file."
+            " Not recommended with multiple conda installations when softlinks are enabled."
+        ),
+    )
+    uninstall_subcommand.add_argument(
+        "--remove-condarcs",
+        choices=["user", "system", "all"],
+        default=None,
+        required=False,
+        help=(
+            "Remove all .condarc files."
+            " `user` removes the files inside the current user's"
+            " home directory and `system` removes all files outside of that directory."
+            " Not recommended when multiple conda installations are on the system"
+            " or when running on an environments directory."
+        ),
+    )
+    uninstall_subcommand.add_argument(
+        "--remove-caches",
```

Comment:
```suggestion
        "--remove-additional-caches",
```

The term is a bit convoluted with `pkgs/` and `pkgs/cache` being taken care of in `conda-clean`, but this argument targets `~/.conda` and `anaconda-client` stuff.

Reply:
Yes, unfortunately, `conda` overloads the term "cache". I changed it to `--remove-conda-caches` because it's more descriptive than `--remove-additional-caches`.

Reply:
I still think that if a user sees both `--conda-clean` and `--remove-conda-caches` might get confused about the difference between both. Maybe the best option is to split it explicitly: `--remove-anaconda-client-data`, `--remove-conda-notices-data`. I'd argue the second one belongs in `conda clean` responsibilities though, so what iffff:

- The `--conda-clean` flag _also_ takes care of conda notices, there's no choice. Eventually we upstream this to `conda`.
- Folks can optionally remove `anaconda-client` data.

Reply:
I'm more amenable to the option that `--clean-caches` removes the notices cache and runs `conda clean` and that `--clean-conda-data` or `--clean-user-data` removes `~/.conda` and `anaconda-client` data.

Reply:
The more I think about this (and after talking to a conda maintainer) the more convinced I am that we shouldn't handle `anaconda-client` data here. This should be handled by the package instead. Otherwise, we would have to implement clean-up operations for other packages or an arbitrary selection.

I think `conda-standalone` should only handle `conda` files. So, my final proposal is:

* `--clean-caches` to run `conda clean` and remove the notices cache.
* `--remove-user-data` to just delete `~/.conda` for now.

This also follows somewhat the `XDG_*` convention where we have cache, data, and configs (here `.condarc` files). In that case, I can also rename `--remove-condarcs` to `--remove-config-files` to allow `constructor` to use more generic terms.

What do you think about this solution?

Reply:
Love this last proposal and the XDG rationale. I think we should stick to a verb only: either `clean`, `clear` or `remove`, but not a mix. So pick one (using `clean` as an example) and then we say:

- `--clean-caches`
- `--clean-user-data`
- `--clean-config-files`

Reply:
I prefer `remove-*` because it tells the user that files will be deleted.